nn.js

```diff
@@ -45,27 +49,33 @@ export class NeuralNetwork {
     }
     train(input, target) {
         for (let i = 0; i < this.epochs; i++) {
+            // Feed forward
             let input_layer = input;
-            let hidden_layer = multiply(input_layer, this.synapse0).map(v => this.activation(v, false));
-            let output_layer = multiply(hidden_layer, this.synapse1).map(v => this.activation(v, false));
-
+            let hidden_layer_linear = multiply(input_layer, this.synapse0)
+            let hidden_layer_activated = hidden_layer_linear.map(v => this.activation(v, false));
+            let output_layer_linear = multiply(hidden_layer_activated, this.synapse1)
+            let output_layer_activated = output_layer_linear.map(v => this.activation(v, false));
+            // Compute output error
             let output_error = subtract(target, output_layer);
-            let output_delta = dotMultiply(output_error, output_layer.map(v => this.activation(v, true)));
+            // Backpropagation of the output error to the output layer 
+            let output_delta = dotMultiply(output_error, multiply(hidden_layer_activated, this.weight1).map(v => this.activation(v, true)));            
```

Comment:
Why do you multiple `hidden_layer_activated` two times? This is already activated, just get derivative from it